docs/write-audit-log-filter-definitions.md

```diff
@@ -0,0 +1,760 @@
+
+- JSON-based filter rules are flexible: you can define per-class, per-subclass, per-user rules.  
+- Log rotation and pruning can be automatic (via global variables) or manual (via functions like audit_log_rotate()). 
+
+- Encryption operations are optional and only needed if audit_log_encrypt is enabled. 
+
+- Combining operations + variables allows full control: define rules, assign users, rotate logs, prune old data, and inspect activity.  
+
+### Filtering approaches
+* Inclusive filtering - Log only specific events you want to track
+
+* Exclusive filtering - Log everything except events you want to exclude
+
+* Mixed filtering - Combine inclusion and exclusion rules
+
+## Basic structure
+
+A filter consists of a top‑level object named `filter`.
+
+Inside this object you can set a global `log` flag and optionally define an array called `class`.
+
+Each element of the `class` array identifies a class name (for example, `connection`, `general`, `query`, `table_access`). 
+
+Each class element may contain an `event` object that specifies particular events and whether they should be logged.
+
+You can also specify which user accounts the filter applies to using the `user` field within each class element. This allows you to include or exclude specific accounts from audit logging.
+
+```json
+{
+  "filter": {
+    "log": true,
+    "class": [
+      {
+        "name": "connection",
+        "event": {
+          "name": "connect",
+          "log": true
+        }
+      },
+      {
+        "name": "query",
+        "event": {
+          "name": "execute",
+          "log": true
+        }
+      }
+    ]
+  }
+}
+```
+
+If the `log` flag is omitted, the default value is `true`.
+When `log` is set to `false`, the filter disables logging for all statement‑type events unless a class entry overrides that setting.
+
+## Filter result values
+
+Understanding the values used in filter definitions:
+
+### Log values
+* `"log": true` - Enable logging for this event or class
+* `"log": false` - Disable logging for this event or class
+* When omitted - Defaults to `true` (logging enabled)
+
+### Status values
+* `"status": [0]` - Log only successful operations
+* `"status": [1]` - Log only failed operations  
+* `"status": [0, 1]` - Log both successful and failed operations
+
+### Negate values
+* `"negate": true` - Exclude the specified users/events from logging
+* `"negate": false` or omitted - Include the specified users/events in logging
+
+### Operation values
+* `"operation": ["connect", "disconnect"]` - Log specific connection events
+* `"operation": ["insert", "update", "delete"]` - Log specific table access operations
+* `"operation": ["select"]` - Log only SELECT queries
+
+## Practical example
+
+Suppose you want a user named `excludeUserTest` to generate no audit records for statements, but the server should keep a generic record that a connection was established.
```

Comment:
The filter below will record no events (connection events will also be excluded).